### PR TITLE
Fixes of AI flow & prompts

### PR DESCRIPTION
**Pull request by abrarbarbhuyia**
- Fixes to mealingredients (checks before adding mealid to array)
- Edit prompts to make menu & ingredient items more consistent
- Increased mongodb timeout to prevent topology error

### CODE REVIEW

**Review by gbashour — APPROVED** [review]

File: server/allergenMaster.js [hunk 1]
```diff
@@ -20,17 +21,21 @@ async function getMenuImage(body) {
...
+  const prompt = `In only a JSON object response, provide the ingredients for ${query}, inside each ingredient, map a list of allergens to each ingredient. Given these allergens come from the following list ${dietaryRequirements}. Do not include your answer inside a string, just a JSON Code block. Be very accurate with the ingredients you list, ensuring to take inspiration from the dish name. Do not list the amount of ingredients, simply identify them. Do not include more than 25 ingredients per meal. Structure the JSON so that it is exactly like {ingredients: [{name:"", allergens:[]}, and so on]} RETURN A JSON AND PUT ALL THIS UNDER INGREDIENTS:`;
+  await new Promise(resolve => setTimeout(resolve, 10000));
   return callGeminiJSON(prompt);
+  // return callGPTJSON(prompt);
```
Comment (gbashour) [inline server/allergenMaster.js:38]: Why is this return commented out if we're going with the old one?

Comment (abrarbarbhuyia) [reply server/allergenMaster.js:38]: idk

Comment (abrarbarbhuyia) [reply server/allergenMaster.js:38]: For quick switching between our two models! 😁

Comment (AdamNas123) [reply server/allergenMaster.js:38]: Why did you bother to comment in the first place

**Pull request merged**